From f4d4be919f70b0372936a0820365f68a79e08827 Mon Sep 17 00:00:00 2001
From: Satish Kumar Nagireddy <satish.nagireddy.nagireddy@xilinx.com>
Date: Tue, 5 Mar 2019 17:54:09 -0800
Subject: [PATCH 0537/1852] dma: xilinx: Define early callback modes

commit ede356c53e69b231100b1800d9aaa7bdd2bc3993 from
https://github.com/Xilinx/linux-xlnx.git

This patch adds two early callback flags in frmbuf driver. There are
four possible stages of buffer descriptors called pending, staged, active
and done. EARLY_CALLBACK is to indicate the driver to release the buffer
after staged stage. EARLY_CALLBACK_LOW_LATENCY is to indicate the driver
to release the buffer before the actual DMA transfer start.

Signed-off-by: Satish Kumar Nagireddy <satish.nagireddy.nagireddy@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 include/linux/dma/xilinx_frmbuf.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/include/linux/dma/xilinx_frmbuf.h b/include/linux/dma/xilinx_frmbuf.h
index 3587998da102..67848abbcf15 100644
--- a/include/linux/dma/xilinx_frmbuf.h
+++ b/include/linux/dma/xilinx_frmbuf.h
@@ -14,6 +14,10 @@
 
 #include <linux/dmaengine.h>
 
+/* Modes to enable early callback */
+#define EARLY_CALLBACK			BIT(1) /* To avoid first frame delay */
+#define EARLY_CALLBACK_LOW_LATENCY	BIT(2) /* Low latency mode */
+
 /**
  * enum vid_frmwork_type - Linux video framework type
  * @XDMA_DRM: fourcc is of type DRM
-- 
2.31.1

